### PR TITLE
input component

### PR DESCRIPTION
**Pull request by vincentv**

### CODE REVIEW

File: src/components.d.ts [hunk 1]
```diff
@@ -58,6 +59,50 @@ export namespace Components {
          */
         "name": string;
     }
+    interface P6Input {
+        "disabled": boolean;
```
Comment (jswale) [inline src/components.d.ts:63]: Missing comment like the other fields

File: src/components.d.ts [hunk 2]
```diff
@@ -58,6 +59,50 @@ export namespace Components {
...
+        /**
+          * The maximum length or value
+         */
+        "max": string;
+        /**
+          * The minimum length or value
+         */
+        "min": string;
+        "multiline": boolean;
```
Comment (jswale) [inline src/components.d.ts:72]: Ditto

File: src/components/atoms/p6-input/p6-input.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+   * shows a waiting indicator
+   */
+  @Prop() waiting: boolean = false;
+
+  @State() hasError: boolean = false;
+
+  private nativeInput: HTMLInputElement | HTMLTextAreaElement | undefined;
+
+  @Listen("focusout")
+  lostFocusHandler() {
```
Comment (jswale) [inline src/components/atoms/p6-input/p6-input.tsx:78]: Can you DRY lostFocusHandler and componentDidLoad code?

File: src/utils/attribute.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+
+function isUndefined(value: unknown): boolean {
+    return value === undefined 
+    || value === "undefined"
+    || value === null 
+    || value === "null"
+    || `${value}`.trim() === ''
```
Comment (jswale) [inline src/utils/attribute.ts:7]: an empty/blank string is not an undefined value for me...

File: src/components/atoms/p6-input/p6-input.tsx (hunk 3 above)
Comment (vincentv) [reply src/components/atoms/p6-input/p6-input.tsx:78]: done

File: src/components.d.ts (hunk 1 above)
Comment (vincentv) [reply src/components.d.ts:63]: It's a stencil-generated file.

**Pull request merged**